timemachine/fe/free_energy.py

```diff
@@ -268,6 +278,38 @@ def frames(self) -> List[StoredArrays]:
     def boxes(self) -> List[NDArray]:
         return [np.array(traj.boxes) for traj in self.trajectories]
 
+    def compute_u_kn(self) -> Tuple[NDArray, NDArray]:
+        """get MBAR input matrices u_kn and N_k"""
+
+        # TODO: there's probably a much faster idiom for this -- shouldn't take a whole minute...
+        N_k = [len(traj.frames) for traj in self.trajectories]
+        N, K = sum(N_k), len(N_k)
+        assert len(self.final_result.initial_states) == K
+
+        kBTs = [BOLTZ * state.integrator.temperature for state in self.final_result.initial_states]
+
+        # TODO: validate assumption that all states are in compatible thermodynamic states
```

Comment:
Could this just reuse https://github.com/proteneer/timemachine/blob/80e4aae47169688200a802612c48cc54a0ee1010/timemachine/fe/free_energy.py#L1164?
(and also `assert_potentials_compatible`?)

Reply:
Good call -- applied in https://github.com/proteneer/timemachine/pull/1387/commits/cfbfead6616231a4e1ad636b5f66b60a4e641653